system-int-test/test/cmr/system_int_test/ingest/variable_ingest_with_association_test.clj

```diff
@@ -214,3 +214,78 @@
+                                                                                           :ShortName "S2"}))
+        coll2-PROV1-2 (data-core/ingest-umm-spec-collection "PROV1" (data-umm-c/collection {:EntryTitle "E2"
+                                                                                            :ShortName "S2"}))
+        _ (index/wait-until-indexed)]
+
+    (testing "ingest of a new variable concept with association on PROV1"
+      (let [concept (variable-util/make-variable-concept
+                     {:Dimensions [(umm-v/map->DimensionType {:Name "Solution_3_Land"
+                                                              :Size 3
+                                                              :Type "OTHER"})]
+                      :AcquisitionSourceName "Instrument1"}
+                     {:native-id "var1"
+                      :coll-concept-id (:concept-id coll1-PROV1-1)})
+            {:keys [concept-id revision-id variable-association]}
+              (variable-util/ingest-variable-with-association concept)
+            var-concept-id concept-id
+            va-concept-id-1 (:concept-id variable-association)
+            va-revision-id-1 (:revision-id variable-association)]
+        (is (mdb/concept-exists-in-mdb? concept-id revision-id))
+        (is (= 1 revision-id))
+        (is (mdb/concept-exists-in-mdb? va-concept-id-1 va-revision-id-1))
+        (is (= 1 va-revision-id-1))
+
+        (testing  "update the variable above with the same collection with revision on PROV1 is OK"
+          (let [concept (variable-util/make-variable-concept
+                          {:Dimensions [(umm-v/map->DimensionType {:Name "Solution_3_Land"
+                                                                   :Size 3
+                                                                   :Type "OTHER"})]
+                           :AcquisitionSourceName "Instrument1"}
+                          {:native-id "var1"
+                           :coll-concept-id (:concept-id coll1-PROV1-1)
+                           :coll-revision-id (:revision-id coll1-PROV1-1)})
+                {:keys [concept-id revision-id variable-association]}
+                  (variable-util/ingest-variable-with-association concept)
+                va-concept-id-2 (:concept-id variable-association)
+                va-revision-id-2 (:revision-id variable-association)]
+            (is (mdb/concept-exists-in-mdb? concept-id revision-id))
+            (is (= 2 revision-id))
+            (is (mdb/concept-exists-in-mdb? va-concept-id-2 va-revision-id-2))
+            (is (= 1 va-revision-id-2))
+            (is (not= va-concept-id-2 va-concept-id-1))
+
+            ;; Verify va-concept-1 is deleted.
+            (is (= true (:deleted (mdb/get-concept va-concept-id-1))))
+
+            (testing  "update the variable above with different collection on PROV1 is OK"
```

Comment:
I don't like this update feature. Having user specify a `native-id` which can silently update any variables on the same provider with the same native-id is very risky. That is a lot of trust to put on the client user to manage the native-id correctly. I wouldn't trust myself to keep track of the native-ids that I have used.